x-pack/plugins/security/server/routes/authentication.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { RouteDefinitionParams } from '.';
+import { SAMLLoginStep } from '../authentication';
+
+export function defineAuthenticationRoutes(params: RouteDefinitionParams) {
+  if (params.config.authc.providers.includes('saml')) {
+    defineSAMLRoutes(params);
+  }
+}
+
+/**
+ * Defines routes required for SAML authentication.
+ */
+function defineSAMLRoutes({ basePath, router, logger, authc }: RouteDefinitionParams) {
+  router.get(
+    {
+      path: '/api/authc/saml/capture-current-url',
+      validate: { query: schema.object({ currentPath: schema.string() }) },
+      options: { authRequired: false },
+    },
+    (context, request, response) => {
+      const { currentPath } = request.query;
+      return response.custom({
+        body: `
+        <!DOCTYPE html>
+        <title>Kibana SAML Login</title>
+        <script>
+          window.location.replace(
+            '${basePath.get(
+              request
+            )}/security/api/authc/saml/start?currentURL=' + encodeURIComponent('${currentPath}' + window.location.hash)
```

Comment:
`currentPath` looks like it can be any user-controlled string, which would make this an XSS vector?

Reply:
(e.g. provide a `currentPath` along the lines of `')); alert('hi!')//`

Reply:
> currentPath looks like it can be any user-controlled string, which would make this an XSS vector?

Yep, it's exactly what you see here, glaring XSS vulnerable code (with a bit tweaked payload) :slightly_smiling_face: We're just not sure where to place this `currentPath` yet, initially we stored it in the intermediate encrypted cookie so that we don't need to do a "is it a relative path" check and care about proper escaping. If we still decide to keep it here I'll take care of properly handling this including the "is it a relative path"  check.

Thank you for being on guard!